emcc.py

```diff
@@ -2200,6 +2199,21 @@ def do_minify(): # minifies the code. this is also when we do certain optimizati
               os.unlink(js_target) # we don't need the js, it can just confuse
               os.unlink(asm_target) # we don't need the asm.js, it can just confuse
             sys.exit(0) # and we are done.
+        if opt_level >= 2:
```

Comment:
It's basically the same. The difference is that for asm.js, we do it in js_optimizer.py, which first splits out the asm.js, then can run passes on that code separately from the shell JS. It also coordinates those opts, as some do affect the shell, so there is a bit of complexity there. I didn't see a natural way to share this code. Instead, my hope is that at some point we'll deprecate asm.js support (leaving it to older versions), and get rid of the js_optimizer code path.